tests/components/gardena_bluetooth/__init__.py

```diff
@@ -59,3 +70,18 @@
     service_uuids=["98bd0001-0b0e-421a-84e5-ddbf75dc6de4"],
     source="local",
 )
+
+
+async def setup_entry(
+    hass: HomeAssistant, mock_entry: MockConfigEntry, platforms: list[Platform]
+) -> Coordinator:
+    """Make sure the device is available."""
+
+    inject_bluetooth_service_info(hass, WATER_TIMER_SERVICE_INFO)
+
+    with patch("homeassistant.components.gardena_bluetooth.PLATFORMS", platforms):
+        mock_entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+    return hass.data[DOMAIN][mock_entry.entry_id]
```

Comment:
Don't use anything stored in hass.data in the tests. We want to avoid interacting integration details.

Patch and mock the library and modify time as needed and assert core state and library calls.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
https://github.com/home-assistant/core/pull/96921